spec/index.bs

```diff
@@ -2036,12 +2043,150 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
 
 # Pipelines # {#pipelines}
 
+## Base pipelines # {#pipeline-base}
+
 <script type=idl>
 dictionary GPUPipelineDescriptorBase : GPUObjectDescriptorBase {
-    required GPUPipelineLayout layout;
+    GPUPipelineLayout layout;
+};
+
+interface mixin GPUPipelineBase {
+    GPUBindGroupLayout getBindGroupLayout(unsigned long index);
 };
 </script>
 
+{{GPUPipelineBase}} has the following internal slots:
+<dl dfn-type=attribute dfn-for="GPUPipelineBase">
+    : <dfn>\[[layout]]</dfn> of type `GPUPipelineLayout`.
+    ::
+        The definition of the layout of resources which can be used with `this`.
+</dl>
+
+### <dfn method for=GPUPipelineBase>getBindGroupLayout(index)</dfn> ### {#pipelinebase-getBindGroupLayout}
+
+<div algorithm="GPUPipelineBase.getBindGroupLayout">
+    **Arguments:**
+
+        - {{unsigned long}} |index|
+
+    **Returns:** {{GPUBindGroupLayout}}
+
+    1. If |index| is greater or equal to {{GPULimits/maxBindGroups}}:
+
+        1. Throw a {{RangeError}}.
+
+    1. If |this| is not [=valid=]:
+
+        1. Return a new error {{GPUBindGroupLayout}}.
+
+    1. Return a new {{GPUBindGroupLayout}} wrapping
```

Comment:
the `|this|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayout]]}}[|index|]` already has a type of `GPUBindGroupLayout`. What is the "wrapping" here?

Reply:
I wanted to say that a new JS object is created so that `pipeline.getBindGroupLayout(i)` always provide the same JS object but at the same time have it be unrelated to `GPUBindGroupLayout` objects passed in `GPUPipelineLayoutDescriptor.bindGroupLayouts`.

This is to avoid having to deal with what happens if you call `getBindGroupLayout` on valid vs. invalid pipeline that was created with an explicit pipeline layout. Since the error state is on the device timeline, we don't know it in the call to `getBindGroupLayout` that's in the content timeline, and can't know if we should return the same `GPUBindGroupLayout` that was passed explicitly via that `GPUPipelineLayout` (with its expando properties) or a new JS object that represents an error `GPUBindGroupLayout`.

Reply:
Ok, that makes sense.
Sounds like some clarification is needed in the spec itself. Maybe shorter, but just saying it's a "wrap" is not clear.

Reply:
I'm not 100% sure how to phrase that, I wanted to write `a new {{GPUBindGroupLayout}} that references the same [=bind group layout=] as ...` but `[=object=]` only exists for device and adapter. Is the plan to do it for all objects? +CC @kainino0x

Reply:
We can leave this as is and then fix it to say "new {{GPUBindGroupLayout}} that is [=group-equivalent=] to ..." after #724 lands

Reply:
> Is the plan to do it for all objects? +CC @kainino0x

Yes (see "define texture (internal object)", "define buffer (internal object)")